src/persist-client/src/internal/state_versions.rs

```diff
@@ -329,6 +329,26 @@ impl StateVersions {
                 shard_metrics
                     .inline_part_bytes
                     .set(u64::cast_from(size_metrics.inline_part_bytes));
+
+                {
+                    // Capture and report spine-batch-specific metrics.
+                    let mut compact_batches = 0;
+                    let mut compacting_batches = 0;
+                    let mut noncompact_batches = 0;
+                    for batch in new_state.collections.trace.spine_batches() {
```

Comment:
Could we do a `size_metrics` type pattern, where some fn returns these metrics as a struct instead of exposing spine batches externally to Trace? I'd prefer that if so

Reply:
Sure - I'll roll that in.

Reply:
(I think the odds that we end up wanting to expose spine batches outside the file for other reasons are ~significant, but happy to kick that can down the road a bit!)